File: RecordWorkouts/views/ShowMoreTableCell.m
#import "ShowMoreTableCell.h"
#import "../WRFormat.h"

@implementation ShowMoreTableCell {
    void (^_callbackHandler)(void);
}

static UIColor *defaultTextColor;

@synthesize dateLabel;
@synthesize showMoreButton;

+ (void) initialize {
    defaultTextColor = [UIColor colorWithRed:210.0f/255.0f
                                       green:210.0f/255.0f
                                        blue:210.0f/255.0f
                                       alpha:1.0f];
}

- (void)awakeFromNib {
    [super awakeFromNib];
}

- (void) setQueryDate: (NSDate *)queryFromDate onShowMore: (void (^)(void)) onShowMore; {
    [self setTextOn:dateLabel text:[WRFormat formatDate:queryFromDate] size:12];
    _callbackHandler = onShowMore;
}

- (IBAction) onShowMoreClick: (id)sender {
    if(_callbackHandler) {
        _callbackHandler();
    }
}

- (void)setTextOn:(UILabel *)lbl text:(NSString *)text size:(int)size {
    lbl.font = [UIFont fontWithName:@"Verdana" size:size];
    [lbl setTextColor:defaultTextColor];
    [lbl setText:text];
}

@end

File: workout-records/views/ShowMoreTableCell.swift
import UIKit

class ShowMoreTableCell: UITableViewCell {
    private static var defaultTextColor = UIColor(named: "showMore_text")
    private var callbackHandler: (() -> Void)?
    
    @IBOutlet var dateLabel: UILabel!
    
    override func awakeFromNib() {
        super.awakeFromNib()
    }
    
    func setQueryDate(_ queryFromDate: Date, onShowMore: @escaping () -> Void) {
        setTextOn(dateLabel, text: WRFormat.formatDate(queryFromDate), size: 14)
        callbackHandler = onShowMore
    }
    
    @IBAction func onShowMoreClick(_ sender: Any) {
        if let cb = callbackHandler { cb() }
    }
    
    private func setTextOn(_ lbl: UILabel, text: String, size: Int) {
        lbl.font = UIFont(name: "Verdana", size: CGFloat(size))
        lbl.textColor = ShowMoreTableCell.defaultTextColor
        lbl.text = text
    }
}
